utility/src/test/java/org/oppia/android/util/logging/performancemetrics/ProdPerformanceMetricsUtilsTest.kt

```diff
@@ -38,95 +43,126 @@ import org.oppia.android.util.platformparameter.SPLASH_SCREEN_WELCOME_MSG_DEFAUL
 import org.oppia.android.util.platformparameter.SYNC_UP_WORKER_TIME_PERIOD_IN_HOURS_DEFAULT_VALUE
 import org.oppia.android.util.platformparameter.SplashScreenWelcomeMsg
 import org.oppia.android.util.platformparameter.SyncUpWorkerTimePeriodHours
-import org.robolectric.RobolectricTestRunner
-import org.robolectric.RuntimeEnvironment
 import org.robolectric.Shadows.shadowOf
 import org.robolectric.annotation.Config
 import org.robolectric.annotation.LooperMode
 import org.robolectric.shadows.ShadowActivityManager
 import java.io.File
 import javax.inject.Inject
+import javax.inject.Qualifier
 import javax.inject.Singleton
 
 private const val TEST_PACKAGE_NAME = "TEST_PACKAGE_NAME"
 private const val TEST_PACKAGE_LABEL = "TEST_PACKAGE_LABEL"
 private const val TEST_APP_PATH = "TEST_APP_PATH"
 private const val TEST_PID = 1
+private const val TEST_UID = 1
 
-/** Tests for [PerformanceMetricsUtils]. */
+/** Tests for [ProdPerformanceMetricsUtils]. */
 // FunctionName: test names are conventionally named with underscores.
 @Suppress("FunctionName")
-@RunWith(RobolectricTestRunner::class)
+@RunWith(AndroidJUnit4::class)
 @LooperMode(LooperMode.Mode.PAUSED)
-@Config(application = PerformanceMetricsUtilsTest.TestApplication::class)
-class PerformanceMetricsUtilsTest {
+@Config(application = ProdPerformanceMetricsUtilsTest.TestApplication::class)
+class ProdPerformanceMetricsUtilsTest {
 
   @Inject
-  lateinit var performanceMetricsUtils: PerformanceMetricsUtils
+  lateinit var prodPerformanceMetricsUtils: ProdPerformanceMetricsUtils
 
-  @Inject
-  lateinit var context: Application
+  @field:[Inject MockContext]
+  lateinit var mockContext: Context
 
   @Before
   fun setUp() {
+    MockitoAnnotations.initMocks(this)
     setUpTestApplicationComponent()
   }
 
   @Test
   fun testPerformanceMetricsUtils_setTotalMemory_returnsCorrectMemoryTier() {
     val activityManager: ActivityManager =
-      RuntimeEnvironment.application.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+      ApplicationProvider.getApplicationContext<Application>()
+        .getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+    `when`(mockContext.getSystemService(Context.ACTIVITY_SERVICE)).thenReturn(activityManager)
```

Comment:
We should never be mocking Android classes directly. Can we use/create shadows to do what's needed here, instead?

See: http://robolectric.org/best-practices/#:~:text=DON'T%20mock%20or%20spy,Robolectric%20or%20Android%20Platform%20upgrades.

Reply:
got it, made amends.